Content.Server/_DV/Containers/ContentContainerSystem.cs

```diff
@@ -0,0 +1,52 @@
+using System.Linq;
+using Content.Server.Polymorph.Systems;
+using Content.Shared.Mind.Components;
+using Robust.Server.Containers;
+using Robust.Server.GameObjects;
+using Robust.Shared.Containers;
+
+namespace Content.Server._DV.Containers;
+
+public sealed class ContentContainerSystem : EntitySystem
+{
+    [Dependency] private readonly ContainerSystem _container = default!;
+    [Dependency] private readonly TransformSystem _transform = default!;
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<ContainerManagerComponent, BeforePolymorphedEvent>(OnBeforePolymorphed);
+
+        base.Initialize();
+    }
+
+    private void OnBeforePolymorphed(Entity<ContainerManagerComponent> ent, ref BeforePolymorphedEvent args)
+    {
+        // Recursively drop all entities with MindContainerComponent in all containers on the entity
+        // This prevents players from being sent to null-space when carried by a polymorphing entity
+        var stack = new Stack<EntityUid>();
+        stack.Push(ent);
+
+        while (stack.Count > 0)
+        {
+            var currentUid = stack.Pop();
+
+            if (!HasComp<ContainerManagerComponent>(currentUid))
+                continue;
+
+            foreach (var container in _container.GetAllContainers(currentUid).ToList())
+            {
+                foreach (var entity in container.ContainedEntities.ToList())
```

Comment:
remove ToList from these bruh

defer the actual unparenting to after the recursion is done:
1. have a reused `private List<EntityUid> _found = new();` on the system
2. call `_found.Clear();` at the start of this
3. use `_found.Add(entity);` instead of attaching
4. at the end `foreach (var entity in _found) { _transform.AttachToGridOrMap(entity); }`